src/query/interpret/eval.hpp

```diff
@@ -258,7 +327,17 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
   }
 
   TypedValue Visit(PropertyLookup &property_lookup) override {
-    auto expression_result = property_lookup.expression_->Accept(*this);
+    ReferenceExpressionEvaluator referenceExpressionEvaluator(frame_, symbol_table_, ctx_);
+
+    TypedValue *expression_result = nullptr;
+    expression_result = property_lookup.expression_->Accept(referenceExpressionEvaluator);
+
+    TypedValue expression_result_;
```

Comment:
Usually we only use the `_` postfix for private member variables. I think the same schema that you did with lists is beter (`lhs_ptr` and `lhs`, here it should be `expression_result_ptr` and `expression_result`).

Reply:
True, true. Fixed